src/otx/algorithms/visual_prompting/tasks/inference.py

```diff
@@ -258,7 +310,76 @@ def export(  # noqa: D102
+        For this case, SAM should be divided into two parts, image encoder and prompt encoder + mask decoder.
+
+        Args:
+            export_type (ExportType): Export type should be ExportType.OPENVINO
+            output_model (ModelEntity): The model entity in which to write the OpenVINO IR data
+            precision (bool): Output model weights and inference precision
+            dump_features (bool): Flag to return "feature_vector" and "saliency_map".
+
+        Raises:
+            Exception: If export_type is not ExportType.OPENVINO
+        """
+        if dump_features:
+            logger.warning(
+                "Feature dumping is not implemented for the visual prompting task."
+                "The saliency maps and representation vector outputs will not be dumped in the exported model."
+            )
+
+        if export_type == ExportType.ONNX:
+            output_model.model_format = ModelFormat.ONNX
+            output_model.optimization_type = ModelOptimizationType.ONNX
+            if precision == ModelPrecision.FP16:
+                raise RuntimeError("Export to FP16 ONNX is not supported")
+        elif export_type == ExportType.OPENVINO:
+            output_model.model_format = ModelFormat.OPENVINO
+            output_model.optimization_type = ModelOptimizationType.MO
+        else:
+            raise RuntimeError(f"not supported export type {export_type}")
+
+        self.precision[0] = precision
+        output_model.has_xai = dump_features
+
+        logger.info("Exporting to the OpenVINO model.")
+        onnx_path = {
+            "visual_prompting_image_encoder": os.path.join(self.output_path, "visual_prompting_image_encoder.onnx"),
+            "visual_prompting_decoder": os.path.join(self.output_path, "visual_prompting_decoder.onnx"),
+        }
+        self._export_to_onnx(onnx_path)
+
+        if export_type == ExportType.ONNX:
+            for module, path in onnx_path.items():
+                with open(path, "rb") as file:
+                    output_model.set_data(f"{module}.onnx", file.read())
+        else:
+            for module, path in onnx_path.items():
+                optimize_command = [
+                    "mo",
+                    "--input_model",
+                    path,
+                    "--output_dir",
+                    self.output_path,
+                    "--model_name",
+                    module,
+                ]
+                if precision == ModelPrecision.FP16:
+                    optimize_command.append("--compress_to_fp16")
+                subprocess.run(optimize_command, check=True)
+                with open(path.replace(".onnx", ".bin"), "rb") as file:
+                    output_model.set_data(f"{module}.bin", file.read())
+                with open(path.replace(".onnx", ".xml"), "rb") as file:
+                    output_model.set_data(f"{module}.xml", file.read())
```

Comment:
I think if we set name as openvino.xml and openvino.bin, we can test without modifying test. Correct me if I'm wrong

Reply:
You're right. But visual prompting model should be divided into image encoder and decoder (prompt encoder + mask decoder) because they run independently.
Image encoder runs just once per image, but decoder runs the number of prompts times and this number of times is various.
If you can recommend any idea to combine these two models, I'll update it in other PR.

Reply:
Ah, I didn't know about that. It will be better if we can combine them together, but it seems hard job to do.